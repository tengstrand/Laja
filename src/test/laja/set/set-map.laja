#write "target/test-output/set-map.out"
  #set (x = {
    a: 123
    b: ["h", "i"]
    c: [{ x1: "value" 
          x2: { 
	        a1: x.a 
		    a2: "name" 
          }
        }, {y1: "age"}]
    d: {}
    e: []
  })
  x = {x}
  x.b = {x.b}
  x.b[0] = {x.b[0]}
  x.b[1] = {x.b[1]}
  x.c[0] = {x.c[0]}
  x.c[0].x1 = {x.c[0].x1}
  x.c[0].x2.a1 = {x.c[0].x2.a1}
  x.c[0].x2.a2 = {x.c[0].x2.a2}
  x.c[1].y1 = {x.c[1].y1}
#end
