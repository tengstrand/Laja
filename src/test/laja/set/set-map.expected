  x = {a=123, b=[h, i], c=[{x1=value, x2={a1=123, a2=name}}, {y1=age}], d={}, e=[]}
  x.b = [h, i]
  x.b[0] = h
  x.b[1] = i
  x.c[0] = {x1=value, x2={a1=123, a2=name}}
  x.c[0].x1 = value
  x.c[0].x2.a1 = 123
  x.c[0].x2.a2 = name
  x.c[1].y1 = age
